internal/cmdsearchreplace/cmdsearchreplace_test.go

```diff
@@ -0,0 +1,453 @@
+package cmdsearchreplace
+
+import (
+	"bytes"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"sigs.k8s.io/kustomize/cmd/config/ext"
+	"sigs.k8s.io/kustomize/kyaml/copyutil"
+)
+
+func TestSearchCommand(t *testing.T) {
+	var tests = []struct {
+		name              string
+		input             string
+		args              []string
+		out               string
+		expectedResources string
+		errMsg            string
+	}{
+		{
+			name: "search replace by value",
+			args: []string{"--by-value", "3", "--put-literal", "4"},
+			out: `${baseDir}/
+matched 2 field(s)
+${filePath}:  spec.replicas: 4
```

Comment:
I followed the output format of linux grep which is more intuitive and concise IMO. I felt the output in doc is a bit verbose. WDYT?

Reply:
I wouldn't call this "grep format". The fundamental difference is that grep shows lines verbatim, whereas we're showing a processed view. I think eventually providing a verbatim view with user-defined surrounding context (-c) would be cool (mentioned in future section of the doc), but for now this will suffice.

Anyways, I don't feel strongly about they way you're doing formatting.